frontend/packages/knative-plugin/src/components/add/event-sources/EventSourceSection.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { useFormikContext, FormikValues } from 'formik';
+import * as _ from 'lodash';
+import AppSection from '@console/dev-console/src/components/import/app/AppSection';
+import { FirehoseList } from '@console/dev-console/src/components/import/import-types';
+import CronJobSection from './CronJobSection';
+import SinkBindingSection from './SinkBindingSection';
+import ApiServerSection from './ApiServerSection';
+import KafkaSourceSection from './KafkaSourceSection';
+import YAMLEditorSection from './YAMLEditorSection';
+import { EventSources } from '../import-types';
+import SinkSection from './SinkSection';
+import AdvancedSection from '../AdvancedSection';
+import { isKnownEventSource } from '../../../utils/create-eventsources-utils';
+import ContainerSourceSection from './ContainerSourceSection';
+
+interface EventSourceSectionProps {
+  projects: FirehoseList;
+  namespace: string;
+}
+
+const EventSourceSection: React.FC<EventSourceSectionProps> = ({ projects, namespace }) => {
+  const { values } = useFormikContext<FormikValues>();
+  let EventSource: React.ReactElement;
```

Comment:
Add this line here instead of doing the check in `default` case - `if (!values.type) return null`. We don't want to render anything as the section unless and until a type is selected.

Reply:
done